modules/core/src/v2/coins/xrp.ts

```diff
@@ -301,6 +301,26 @@ export class Xrp extends BaseCoin {
         }
       }
       const id = computeBinaryTransactionHash(txHex);
+
+      if (transaction.TransactionType == 'AccountSet') {
+        return {
+          displayOrder: ['id', 'outputAmount', 'changeAmount', 'outputs', 'changeOutputs', 'fee', 'accountSet'],
+          id: id,
+          changeOutputs: [],
+          outputAmount: 0,
+          changeAmount: 0,
+          outputs: [],
+          fee: {
+            fee: transaction.Fee,
+            feeRate: null,
+            size: txHex.length / 2,
+          },
+          accountSet: {
```

Comment:
Does this property need to be added to the return type?

Reply:
I think callers will need to cast to `any` to use this property, maybe that's ok for now instead of doing some more refactoring.

Reply:
I don't think we actually define the return type anywhere. It seems like the callers know what to expect. One example is the explainTransaction for xlm. 
But we could make a case for defining the type within each coin in account-lib. Couple of things are missing in account-lib like explainTransaction and recover. I have couple of ideas and we could discuss it offline

Reply:
Sounds good. I think callers will need to downcast but until we have a more generic way of extending the return types, this will have to do for now.